concurrency/impl-31/src/main/java/org/jboss/as/ee/concurrent/ConcurroContextServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.ee.concurrent;
+
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+import java.lang.reflect.Proxy;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+
+import static org.jboss.as.ee.logging.EeLogger.ROOT_LOGGER;
+import static org.wildfly.common.Assert.checkArrayBounds;
+import static org.wildfly.common.Assert.checkNotNullParam;
+
+/**
+ * An extension of {@link org.glassfish.concurro.ContextServiceImpl}, which properly supports a security manager.
+ * @author Eduardo Martins
+ */
+public class ConcurroContextServiceImpl extends org.glassfish.concurro.ContextServiceImpl implements WildFlyContextService {
+
+    private final ContextServiceTypesConfiguration contextServiceTypesConfiguration;
+
+    /**
+     *
+     * @param name
+     */
+    public ConcurroContextServiceImpl(String name, ContextServiceTypesConfiguration contextServiceTypesConfiguration) {
+        super(name, new ConcurroDefaultContextSetupProviderImpl(), null);
+        this.contextServiceTypesConfiguration = contextServiceTypesConfiguration;
+    }
+
+    private <T> T internalCreateContextualProxy(T instance, Map<String, String> executionProperties, Class<T> intf) {
+        checkNotNullParam("instance", instance);
+        checkNotNullParam("intf", intf);
+
+        ConcurroIdentityAwareProxyInvocationHandler handler = new ConcurroIdentityAwareProxyInvocationHandler(this, instance, executionProperties);
+        Object proxy = Proxy.newProxyInstance(instance.getClass().getClassLoader(), new Class[]{intf}, handler);
+        return  intf.cast(proxy);
+    }
+
+    @Override
+    public <T> T createContextualProxy(final T instance, final Map<String, String> executionProperties, final Class<T> intf) {
+        if (WildFlySecurityManager.isChecking()) {
```

Comment:
We don't need the security manager support. The security manager has been removed from Jakarta EE 11.